emu_sv/custom_applies.py

```diff
@@ -0,0 +1,22 @@
+import math
+from typing import Any
+
+import torch
+
+from emu_base.base_classes.config import BackendConfig
+from emu_base.base_classes.default_callbacks import QubitDensity
+from emu_base.base_classes.operator import Operator
+
+from emu_sv import StateVector
+
+
+def custom_qubit_density(
+    self: QubitDensity, config: BackendConfig, t: int, state: StateVector, H: Operator
+) -> Any:
+
+    num_qubits = int(math.log2(len(state.vector)))
+    state_tensor = state.vector.reshape((2,) * num_qubits)
```

Comment:
Another idea (maybe for next change) is to provide methods to access either the raw linear vector or the "shaped" tensor, something along the lines of
```
class StateVector:
  def vector_view(self) -> torch.Tensor:
    return self.vector

  def tensor_view(self) -> torch.Tensor:
    return self.vector.reshape((2,) * self.qubit_count)
```

Reply:
I think storing an integer is fine

Reply:
to do later